pennylane/noise/conditionals.py

```diff
@@ -471,46 +624,125 @@ def partial_wires(operation, *args, **kwargs):
     >>> func(qml.RY(1.0, [0]))
     qml.RX(3.2, wires=[0])
 
-    Finally, one can also use ``kwargs`` instead of positional arguments:
+    Moreover, one can use ``kwargs`` instead of positional arguments:
 
     >>> func = qml.noise.partial_wires(qml.RX, phi=1.2)
     >>> func(qml.RY(1.0, [2]))
     qml.RX(1.2, wires=[2])
     >>> rfunc = qml.noise.partial_wires(qml.RX(1.2, [12]), phi=2.3)
     >>> rfunc(qml.RY(1.0, ["light"]))
     qml.RX(2.3, wires=["light"])
+
+    Finally, one may also use this with an instance of
+    :class:`MeasurementProcess <pennylane.measurement.MeasurementProcess>`
+
+    >>> func = qml.noise.partial_wires(qml.expval(qml.Z(0)))
+    >>> func(qml.RX(1.2, wires=[9]))
+    qml.expval(qml.Z(9))
     """
+    is_meas_class = isinstance(operation, qml.measurements.MeasurementProcess)
+    is_meta_class = isinstance(operation, (qml.ops.Adjoint, qml.ops.Controlled))
+    is_mappable, map_op = False, None
+
     if not callable(operation):
         if args:
             raise ValueError(
                 "Args cannot be provided when operation is an instance, "
                 f"got operation = {operation} and args = {args}."
             )
+        if not kwargs:
+            is_mappable, map_op = True, operation
         args, metadata = getattr(operation, "_flatten")()
+        is_metadata_flat = is_meas_class or isinstance(operation, (qml.ops.Controlled))
         if len(metadata) > 1:
-            kwargs = {**dict(metadata[1]), **kwargs}
+            kwargs = {**dict(metadata[1] if not is_metadata_flat else metadata), **kwargs}
         operation = type(operation)
+    elif operation in _MEAS_FUNC_MAP:
+        is_meas_class = True
+        operation = _MEAS_FUNC_MAP[operation]
+    elif operation in (
+        qml.adjoint,
+        qml.ctrl,
+    ):
+        is_meta_class = True
+        operation = {qml.adjoint: qml.ops.Adjoint, qml.ctrl: qml.ops.Controlled}[operation]
 
     fsignature = signature(getattr(operation, "__init__", operation)).parameters
     parameters = list(fsignature)[int("self" in fsignature) :]
     arg_params = {**dict(zip(parameters, args)), **kwargs}
 
-    if "wires" in arg_params:  # Ensure we don't include wires arg
-        arg_params.pop("wires")
+    arg_wires = arg_params.pop("wires", None)  # Ensure we don't include wires in name
+    if is_meas_class and "op" in arg_params:
+        arg_params["obs"] = arg_params.pop("op")
+
+    if is_meta_class and "op" in arg_params:
+        arg_params["base"] = arg_params.pop("op")
 
-    op = partial(operation, **{**arg_params, **kwargs})
+    if operation == qml.ops.Controlled and "control" in arg_params:
+        arg_params["control_wires"] = arg_params.pop("control")
```

Comment:
Why this has to be added now?

Reply:
In order to handle the cases where we have to capture `control` wires for the `Controlled` operations. This comes from the fact that the `ctrl` function uses `control` as the keyword argument but the `Controlled` class uses `control_wires` for it.